cpp/Ice/greeter/Client.cpp

```diff
@@ -11,16 +12,8 @@ using namespace std;
 int
 main(int argc, char* argv[])
 {
-    // Figure out my name.
-    const char* name = getenv("USER");
-    if (name == nullptr)
-    {
-        name = getenv("USERNAME");
-    }
-    if (name == nullptr)
-    {
-        name = "masked user";
-    }
+    // Helper function to get the username of the current user.
```

Comment:
I would get rid of this name variable, we don't use it in other languages.

In the first greet, just call:
```cpp
string greeting = greeter->greet(Env::getUsername());
```